sqlglot/dialects/bigquery.py

```diff
@@ -897,3 +1175,61 @@ def version_sql(self, expression: exp.Version) -> str:
             if expression.name == "TIMESTAMP":
                 expression.set("this", "SYSTEM_TIME")
             return super().version_sql(expression)
+
+        def while_sql(self, expression: exp.While):
+            body = self._sql_statements_block(expression.args.get("body", []))
+            suffix = "\nEND WHILE" if body else "END WHILE"
+            return f"WHILE {self.sql(expression, 'this')} DO\n{body}{suffix}"
```

Comment:
I don't think we should be hardcoding these newlines here. We have `self.sep()` for this, so that you won't have newlines in non-pretty mode.